src/happy.rs

```diff
@@ -0,0 +1,31 @@
+use std::str::FromStr;
+
+use hickory_client::{
+    client::{Client, SyncClient},
+    error::ClientResult,
+    op::DnsResponse,
+    rr::{DNSClass, Name, RecordType},
+    udp::UdpClientConnection,
+};
+
+/*
+TODO:
+This needs to be TCP and have Timeout Retries
+
+*/
+pub fn search(identifier: Name, rtype: RecordType) -> anyhow::Result<DnsResponse> {
+    debug!("searching {:?} for {}", rtype, identifier);
+    let socket_addr = std::net::SocketAddr::new(std::net::IpAddr::from_str("1.1.1.1")?, 53);
+    let client = happy_eyeballs::connect(socket_addr)?;
+    let questionable_addr = client.peer_addr()?;
+    let conn = UdpClientConnection::new(socket_addr).unwrap();
+    let client: SyncClient<UdpClientConnection> = SyncClient::new(conn);
+
+    let q: ClientResult<DnsResponse> = client.query(&identifier, DNSClass::IN, rtype);
+    let response = match q {
+        Ok(response) => response,
+        Err(e) => panic!("{}", e),
+    };
```

Comment:
This is a lot more verbose than it needs to be, and looses any context we might be adding in a different location. I'd suggest this instead:

```suggestion
    let response = client.query(&identifier, DNSClass::IN, rtype)?;
```